tests/test_longitudinal.py

```diff
@@ -18,63 +18,173 @@
-    """Function creates the database state prior to running cupid. To do this
-    we load the generic patient and then add in data that will create the
-    behaviour we want to test.
-
-    Yields:
-        Session: session for test database with some data loaded
-    """
-    xml_test_1 = load_xml_from_path(
-        os.path.join("tests", "xml_files", "store_tests", "test_0.xml")
-    )
-
-    patient_record = PatientRecord(xml_test_1)
-    patient_record.map_to_database(TEST_PID, TEST_UKRDCID, ukrdc_test_session)
-    ukrdc_test_session.add_all(patient_record.get_orm_list())
-
-    xml_test_2 = load_xml_from_path(
-        os.path.join("tests", "xml_files", "store_tests", "test_2.xml")
-    )
-
-    lab_order_start = xml_test_2.lab_orders.start.to_datetime()
-    lab_order_stop = xml_test_2.lab_orders.stop.to_datetime()
-
+def generate_persistent_lab_orders(session:Session, start, stop):
+    # Add records to session that are require for testing laborders
     lab_orders = [
+        # this laborder is within the range so will be deleted as its key would
+        # wouldn't match any of the cupid generated compond keys
         sqla.LabOrder(
             pid=TEST_PID,
             id="to_delete_1",
-            specimencollectedtime=lab_order_start + timedelta(weeks=1),
+            specimencollectedtime=start + timedelta(weeks=1),
         ),
+
+        # this two records should persist as they are outside the range
         sqla.LabOrder(
             pid=TEST_PID,
             id="to_persist_1",
-            specimencollectedtime=lab_order_start - timedelta(weeks=1),
+            specimencollectedtime=start - timedelta(weeks=1),
         ),
         sqla.LabOrder(
             pid=TEST_PID,
             id="to_persist_2",
-            specimencollectedtime=lab_order_stop + timedelta(weeks=1),
+            specimencollectedtime=stop + timedelta(weeks=1),
         ),
+        
+        # this record should be overwritten with laborder in the file with
+        # placerid = 1 if it exists
         sqla.LabOrder(
             pid=TEST_PID,
             id=f"{TEST_PID}:1",
-            specimencollectedtime=lab_order_stop + timedelta(weeks=1),
+            specimencollectedtime=stop + timedelta(weeks=1),
         ),
     ]
-    ukrdc_test_session.add_all(lab_orders)
+    session.add_all(lab_orders)
 
     result_items = [
         sqla.ResultItem(id="RI_to_delete_1", order_id="to_delete_1"),
     ]
-    ukrdc_test_session.add_all(result_items)
-    ukrdc_test_session.commit()
+    session.add_all(result_items)
+    session.commit()
 
-    loaded_ids = [order.id for order in ukrdc_test_session.query(sqla.LabOrder).all()]
+    loaded_ids = [order.id for order in session.query(sqla.LabOrder).all()]
     for order in lab_orders:
         assert order.id in loaded_ids
+    
+    return
+
+
+def generate_persistent_dialysis_sessions(session:Session, start, stop):
+    # Add records to session that are require for testing laborders
```

Comment:
Do the comments on this and generate_persistent_observations need updating to change laborder?

Reply:
Ah you spotted the copy pasted code